src/solvers/dg.jl

```diff
@@ -19,13 +19,59 @@ function get_element_variables!(element_variables, mesh, dg, cache)
-# For instance, shock capturing volume integrals output the blending factor
-# as an "element variable".
-function get_node_variables!(node_variables, mesh, equations,
+# Function to define "node variables" for the SaveSolutionCallback. 
+# Does for hyperbolic equations nothing by default, but can be specialized for certain volume integral types. 
+# For instance, shock capturing volume integrals output the blending factor as a "node variable".
+function get_node_variables!(node_variables, u_ode, mesh, equations,
                              volume_integral::AbstractVolumeIntegral, dg, cache)
-    nothing
+    return nothing
+end
+function get_node_variables!(node_variables, u_ode, mesh, equations,
+                             volume_integral::AbstractVolumeIntegral, dg, cache,
+                             equations_parabolic, cache_parabolic)
+    if !isempty(node_variables)
+        n_nodes = nnodes(dg)
+        n_elements = nelements(dg, cache)
+
+        u = wrap_array(u_ode, mesh, equations, dg, cache)
+        if :vorticity in keys(node_variables)
+            @unpack viscous_container = cache_parabolic
+            @unpack gradients = viscous_container
+            gradients_x, gradients_y = gradients
+
+            vorticity_array = zeros(eltype(cache.elements),
+                                    n_nodes, n_nodes, n_elements)
+
+            @threaded for element in eachelement(dg, cache)
+                for j in eachnode(dg), i in eachnode(dg)
+                    u_node = get_node_vars(u, equations, dg, i, j, element)
+
+                    gradients_1 = get_node_vars(gradients_x, equations_parabolic, dg,
+                                                i, j, element)
+                    gradients_2 = get_node_vars(gradients_y, equations_parabolic, dg,
+                                                i, j, element)
+
+                    vorticity_nodal = vorticity(u_node, (gradients_1, gradients_2),
+                                                equations_parabolic)
+                    vorticity_array[i, j, element] = vorticity_nodal
+                end
+            end
+            node_variables[:vorticity] = vorticity_array
+        end
+    end
+
+    if typeof(volume_integral) == VolumeIntegralSubcellLimiting
+        # While for the element-wise limiting with `VolumeIntegralShockCapturingHG` the indicator is
+        # called here to get up-to-date values for IO, this is not easily possible in this case
+        # because the calculation is very integrated into the method.
+        # See also https://github.com/trixi-framework/Trixi.jl/pull/1611#discussion_r1334553206.
+        # Therefore, the coefficients at `t=t^{n-1}` are saved. Thus, the coefficients of the first
+        # stored solution (initial condition) are not yet defined and were manually set to `NaN`.
+        get_node_variables!(node_variables, volume_integral.limiter, volume_integral,
+                            equations)
+    else
```

Comment:
I agree with you that we should use dispatch here instead of the of clause.

Reply:
Yeah, or we add an if-clause for the hyperbolic case - might be less code, but also less clean? I am not sure what is the better way here.

Reply:
I tend towards using dispatch as then the method can be extended (from outside Trixi.jl) for other types of volume integrals.